submission/proxy_test.go

```diff
@@ -125,15 +142,40 @@ func TestProxyInitState(t *testing.T) {
 	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
 	defer cancel()
 
+Init:
 	for {
 		select {
 		case <-ctx.Done():
 			t.Fatalf("p.Run() expected to send init signal, got none")
-		case b := <-p.Init:
+		case b, ok := <-p.Init:
+			if !ok {
+				t.Fatalf("p.Run() expected to send 'true' init signal via Init channel, but channel is closed")
+			}
 			if b != true {
 				t.Fatalf("p.Run() expected to send 'true' init signal, got false")
 			}
+			break Init
+		case e := <-p.Errors:
+			t.Log(e)
+		}
+	}
+
+	sampleLogListUpdate := strings.Replace(testdata.SampleLogList, "ct.googleapis.com/racketeer/", "ct.googleapis.com/racketeer/v2/", 1)
+	err = updateFile(f, sampleLogListUpdate)
+	if err != nil {
+		t.Fatalf("unable to update Log-list data file: %q", err)
+	}
+	ctx, cancel = context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+	p.llWatcher.RefreshLogListAndNotify(ctx)
+	for {
+		select {
+		case <-ctx.Done():
 			return
+		case _, ok := <-p.Init:
+			if ok {
+				t.Fatalf("p.Refresh() after initial p.Run() invoked init-status signal, expected none")
```

Comment:
A clearer way to say this might be to say explicitly what happened, i.e. that it sent a value to the Init channel. Saying that it "invoked init-status signal" is less clear because the reader has to potentially go away and learn that the Init channel is the "init-status signal".

Reply:
Done

Reply:
![Is it though?](https://media1.tenor.com/images/8a8ceb2c8b436faa23ca7784d5d1d89c/tenor.gif?itemid=13334930)

Reply:
What about now?